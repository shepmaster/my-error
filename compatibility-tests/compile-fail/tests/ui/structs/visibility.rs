// There are also happy-path tests

pub mod inner {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    pub(crate) struct Error;
}

fn private_is_default() {
    let _ = inner::Context.build();
}

fn main() {}
