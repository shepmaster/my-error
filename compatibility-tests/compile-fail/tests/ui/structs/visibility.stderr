error[E0603]: unit struct `Context` is private
  --> $DIR/visibility.rs:11:20
   |
11 |     let _ = inner::Context.build();
   |                    ^^^^^^^ private unit struct
   |
note: the unit struct `Context` is defined here
  --> $DIR/visibility.rs:6:21
   |
6  |     #[derive(Debug, Snafu)]
   |                     ^^^^^
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0624]: associated function `build` is private
  --> $DIR/visibility.rs:11:28
   |
11 |     let _ = inner::Context.build();
   |                            ^^^^^ private associated function
